apps/server/src/shared/infra/identity-management/keycloak-identity-management.ts

```diff
@@ -0,0 +1,154 @@
+import { IAccount, IAccountUpdate } from '@shared/domain';
+
+/* eslint-disable no-nested-ternary */
+import { Inject, Injectable } from '@nestjs/common';
+import KcAdminClient from '@keycloak/keycloak-admin-client';
+import UserRepresentation from '@keycloak/keycloak-admin-client/lib/defs/userRepresentation';
+import { GrantTypes } from '@keycloak/keycloak-admin-client/lib/utils/auth';
+import { IIdentityManagement } from './identity-management.interface';
+
+interface KcSettings {
+	realmName: string;
+	baseUrl: string;
+	credentials: KcCredentials;
+	sourceRealm: string;
+}
+
+interface KcCredentials {
+	grantType: GrantTypes;
+	username: string;
+	password: string;
+	clientId: string;
+}
+
+@Injectable()
+export class KeycloakIdentityManagement implements IIdentityManagement {
+	private lastAuthorizationTime = 0;
+
+	private static AUTHORIZATION_TIMEBOX_MS = 59 * 1000;
+
+	public constructor(
+		@Inject('KeycloakAdminClient') private readonly kcAdminClient: KcAdminClient,
+		@Inject('KeycloakSettings') private readonly kcSettings: KcSettings
+	) {
+		kcAdminClient.setConfig({
+			baseUrl: kcSettings.baseUrl,
+			realmName: kcSettings.realmName,
+		});
+	}
+
+	async createAccount(account: IAccount, password?: string): Promise<string | null> {
+		await this.authorizeAccess();
+		let id: { id: string };
+		try {
+			id = await this.kcAdminClient.users.create({
+				username: account.userName,
+				email: account.email,
+				firstName: account.firstName,
+				lastName: account.lastName,
+			});
+		} catch {
+			return null;
+		}
+
+		if (id && password) {
+			try {
+				await this.resetPassword(id.id, password);
+			} catch {
+				await this.deleteAccountById(id.id);
+				return null;
+			}
+		}
+		return id.id;
+	}
+
+	async updateAccount(accountId: string, account: IAccountUpdate): Promise<string | null> {
+		await this.authorizeAccess();
+		try {
+			await this.kcAdminClient.users.update(
+				{ id: accountId },
+				{
+					email: account.email,
+					firstName: account.firstName,
+					lastName: account.lastName,
+				}
+			);
+			return accountId;
+		} catch {
+			return null;
```

Comment:
The errors here and in the following methods are swallowed. I'd prefer to throw an error.